{"edited":false,"parent_id":"t1_cu20y","id":"cu2fd","author":"nostrademons","author_flair_text":null,"distinguished":null,"retrieved_on":1473808049,"gilded":0,"subreddit_id":"t5_6","stickied":false,"link_id":"t3_tzna","body":"That's what an exception is - an undefined result that happens at runtime, where any further operations on that undefined value result in undefined.  The Haskell exception library makes this explicit.\r\n\r\nAnd in many programming languages, you *can* catch a divide by zero and proceed.  In Java, for example, it throws an ArithmeticException.","controversiality":0,"ups":2,"score":2,"created_utc":1165546578,"author_flair_css_class":null,"subreddit":"reddit.com","timestamp_epoch":1566236762,"message_id":"15184-73a62d1e-9044-4f83-aaf8-06b5a0fb577e-1566236762"}